README.md

```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
 
-### autoBaseDir
+```javascript
+{
+  "driver": ...
+  "plugins": {
+    "view": {
+      "module": "nemo-view"
+    }
+  }
+```
+
+You could also have a config that looks like this, and `nemo-view` will still register itself as `nemo.view`
+
+```javascript
+{
+  "driver": ...
+  "plugins": {
+    "cupcakes": {
+      "module": "nemo-view"
+    }
+  }
+```
+
+But that's confusing. So please stick to the convention.
+
+### Typical usage of Nemo constructor
+
+A typical pattern would be to use `mocha` as a test runner, resolve `nemo` in the context of the mocha `before` function, and use
+the mocha `done` function as the callback:
+
+```javascript
+var nemo;
+describe('my nemo suite', function() {
+  before(function(done) {
+    nemo = Nemo(config, done);
+  });
+  it('will launch browsers!', function(done) {
+    nemo.driver.get('https://www.paypal.com');
+    nemo.driver.quit().then(function() {
+       done();
+    });
+  });
+});
+
+```
+
+## Nemo Configuration
+
+```javascript
+{
+  "driver": { /** properties used by Nemo to setup the driver instance **/ },
+  "plugins": { /** plugins to initialize **/},
+  "data": { /** arbitrary data to pass through to nemo instance **/ }
+}
+```
+
+This configuration object is optional, as long as you've got `nemoData` set as an environment variable (see below).
 
-Used as a root location for finding 'locator' files in your test suite. Also can be required by other plugins
-which need to require modules from your test suite.
+### driver
 
-### targetBrowser
+Here are the `driver` properties recognized by Nemo. This is ALL of them. Please be aware that you really only need to supply "browser" to get things working initially.
 
-Browser you wish to automate. Make sure that your chosen webdriver has this browser option available
+#### browser (optional)
 
-### localServer (optional, defaults to false)
+Browser you wish to automate. Make sure that your chosen webdriver has this browser option available. While this is "optional" you must choose a browser. Either use this property or the "builders.forBrowser" option (see below).
 
-Set localServer to true if you want Nemo to attempt to start a standalone binary on your system (like selenium-standalone-server) or use a local browser/driver like Chrome/chromedriver or PhantomJS.
+#### local (optional, defaults to false)
 
+Set local to true if you want Nemo to attempt to start a standalone binary on your system (like selenium-standalone-server) or use a local browser/driver like Chrome/chromedriver or PhantomJS.
 
-### targetServer (optional)
+
+#### server (optional)
 
 Webdriver server URL you wish to use.
 
-If you are using sauce labs, make sure `targetServer` is set to correct url like `"http://yourkey:yoursecret@ondemand.saucelabs.com:80/wd/hub"`
+If you are using sauce labs, make sure `server` is set to correct url like `"http://yourkey:yoursecret@ondemand.saucelabs.com:80/wd/hub"`
 
```

Comment:
Only setting server with correct sauce url is not enough. Users also must pass `username` and `accessKey` as driver capabilities. Without out it test fails. I tested this and we need to specify this in README

"serverCaps" : {
            "username" : "yourusername",
            "accessKey" : "yourkey"
        }